code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/pushtemplate/PushTemplateTrackers.kt

```diff
@@ -0,0 +1,45 @@
+/*
+  Copyright 2024 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+
+package com.adobe.marketing.mobile.services.ui.pushtemplate
+
+import android.app.Activity
+import android.content.BroadcastReceiver
+
+internal class PushTemplateTrackers private constructor() {
+
+    private var trackerActivities: HashMap<String, Activity> = HashMap()
+    private var broadcastReceivers: HashMap<String, BroadcastReceiver> = HashMap()
```

Comment:
Holding strong references to Android components in static scope is dangerous as it can cause memory leaks if not carefully managed.
Looking at the usages of `Activity` and `BroadcastReceiver` in your PR, it seems to me that the class name of the activity/broadcast receiver is sufficient for the tracking logic to work. Can you check if this approach can be used?

Reply:
i was storing the activity and broadcast receiver to ensure it was created only once then re-used when needed. i did notice that i mistakenly didn't add code to check if the activity or broadcast receiver was already set before creating a new one. the corrected code would look like:
```kotlin
broadcastReceiverName?.let {
            if (PushTemplateTrackers.getInstance().getBroadcastReceiver(broadcastReceiverName) == null) {
                PushTemplateTrackers.getInstance().setBroadcastReceiver(broadcastReceiverName, createClassInstance(context, broadcastReceiverName) as? BroadcastReceiver)
            }
        }
```
what do you think about storing the activity and / or broadcast receiver this way?

Reply:
My question was more on the lines of why an instance of the `Activity` and `BroadcastReceiver` are needed given that the intent for the clicks or deep links only need the class instance to be set on the intent.  Let us discuss more in our pair programming session today.